galley/pkg/config/analysis/analyzers/namespaceconflict/namespaceselector.go

```diff
@@ -0,0 +1,202 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package namespaceconflict
+
+import (
+	"fmt"
+
+	k8s_labels "k8s.io/apimachinery/pkg/labels"
+
+	v1beta1 "istio.io/api/security/v1beta1"
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/pilot/pkg/util/sets"
+	"istio.io/istio/pkg/config/resource"
+	"istio.io/istio/pkg/config/schema/collection"
+	"istio.io/istio/pkg/config/schema/collections"
+)
+
+// Analyzer checks conditions related to conflicting namespace level resources
+type Analyzer struct{}
+
+var _ analysis.Analyzer = &Analyzer{} // TODO: Rename this with simply "Analyzer"
+
+var (
+	peerauthcollection    = collections.IstioSecurityV1Beta1Peerauthentications.Name() // TODO: Match the syntax name of the authpolicy collection, looks cleaner.
+	authpolicycollection  = collections.IstioSecurityV1Beta1Authorizationpolicies.Name()
+	requestauthcollection = collections.IstioSecurityV1Beta1Requestauthentications.Name()
+)
+
+func (a *Analyzer) Metadata() analysis.Metadata {
+	return analysis.Metadata{
+		Name:        "namespaceconflict.Analyzer",
+		Description: "Checks conditions related to Peer Authentication conflicting namespace level resources",
+		Inputs: collection.Names{
+			peerauthcollection,
+			authpolicycollection,
+			requestauthcollection,
+		},
+	}
+}
+
+func (a *Analyzer) Analyze(c analysis.Context) {
+	// Analyze collections.IstioSecurityV1Beta1Peerauthentications
+	c.ForEach(peerauthcollection, func(r *resource.Instance) bool {
+		x := r.Message.(*v1beta1.PeerAuthentication)
+
+		// If the resource has workloads associated with it, analyze for conflicts with selector
+		if x.GetSelector() != nil {
+			a.analyzeWorkloadSelectorConflicts(r, c)
+		}
+		return true
+	})
+
+	// Analyze collections.IstioSecurityV1Beta1Authorizationpolicies
+	c.ForEach(authpolicycollection, func(r *resource.Instance) bool {
+		x := r.Message.(*v1beta1.AuthorizationPolicy)
```

Comment:
I think @yangminzhu mentioned AuthorizationPolicy actually are merged so this may not be applicable for this one. But the other two are

Reply:
Hey @howardjohn @yangminzhu , awesome! I think it makes perfect sense, maybe we can do different type of analysis for that particular resource, I've removed it from this PR.